### ISSUE TITLE
`lifetime may not live long enough` error in `SimpleObject` macro

### ISSUE THREAD

**Issue by adzialocha** [p2panda/aquadoggo#185]
Our CI "sometimes" fails with an undeterministic error in the `SimpleObject` macro of `async-graphql@4.0.4` or Rust compiler:

```
error: lifetime may not live long enough
--> aquadoggo/src/graphql/client/response.rs:12:10
|
12| #[derive(SimpleObject, Serialize, Deserialize, Debug, Eq, PartialEq)]
| -^^^^^^^^^^^
| |
| let's call the lifetime of this reference `'2`
| let's call the lifetime of this reference `'1`
| associated function was supposed to return data with lifetime `'1` but it is returning data with lifetime `'2`
|
= note: this error originates in the derive macro `SimpleObject` (in Nightly builds, run with -Z macro-backtrace for more info)
error: lifetime may not live long enough
--> aquadoggo/src/graphql/client/response.rs:12:10
|
12| #[derive(SimpleObject, Serialize, Deserialize, Debug, Eq, PartialEq)]
| -^^^^^^^^^^^
| |
| let's call the lifetime of this reference `'1`
| let's call the lifetime of this reference `'2`
| associated function was supposed to return data with lifetime `'2` but it is returning data with lifetime `'1`
|
= note: this error originates in the derive macro `SimpleObject` (in Nightly builds, run with -Z macro-backtrace for more info)
error: could not compile `aquadoggo` due to 2 previous errors
```

Related issue: https://github.com/async-graphql/async-graphql/issues/900

**Comment by adzialocha** [comment]
With Version `3.0.38` of the crate and Rust version `1.61.0` it works

**Comment by adzialocha** [comment]
This is fixed now in Rust 1.62.1 https://github.com/rust-lang/rust/issues/98890
